src/main/java/it/gov/pagopa/pu/classification/util/Utilities.java

```diff
@@ -3,21 +3,29 @@
 import org.slf4j.MDC;
 
 import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.text.NumberFormat;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.OffsetDateTime;
 import java.time.temporal.ChronoUnit;
+import java.util.Locale;
 
 public class Utilities {
 
   private Utilities(){}
 
   public static final BigDecimal HUNDRED = BigDecimal.valueOf(100);
+  public static final ThreadLocal<NumberFormat> NUMBER_FORMAT_IT = ThreadLocal.withInitial(() -> NumberFormat.getNumberInstance(Locale.ITALIAN));
```

Comment:
this way you don't need to verify on amountToString  method if it contains the "," with 2 decimals
```suggestion
  public static final ThreadLocal<NumberFormat> DECIMAL_FORMAT = ThreadLocal.withInitial(() -> new DecimalFormat("#,00"));
```